File: zad4.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");
    vector<pair<int, string>> schedule;

    while (true) {
        int N;
        cout << "Введите число операций: ";
        cin >> N;

        string operation;
        for (int i = 0; i < N; ++i) {
            cout << "Введите операцию: ";
            cin >> operation;

            if (operation == "CLASS") {
                string discipline;
                int day;
                cout << "Введите номер дня и название предмета: ";
                cin >> day >> discipline;
                schedule.push_back(make_pair(day, discipline)); // Добавляем занятие в расписание
            } else if (operation == "NEXT") {
                int lastDay = 29; // Предпоследний день текущего месяца

                for (auto& cls : schedule) {
                    cls.first = lastDay; // Переносим все занятия на предпоследний день текущего месяца
                }
                cout << "Перенос всех занятий на предпоследний день текущего месяца выполнен." << endl;
            } else if (operation == "VIEW") {
                int day;
                cout << "Введите номер дня: ";
                cin >> day;

                // Используем алгоритм find_if для поиска первого занятия на указанный день
                auto it = find_if(schedule.begin(), schedule.end(), [day](const pair<int, string>& cls) {
                    return cls.first == day;
                });

                cout << "В " << day << " день ";
                if (it != schedule.end()) { // Если занятия нашлись на указанный день
                    cout << "1 занятие в университете: " << it->second;
                    ++it;
                    while (it != schedule.end() && it->first == day) { // Выводим все занятия на этот день
                        cout << ", " << it->second;
                        ++it;
                    }
                } else {
                    cout << "У нас свободный день!"; // Если занятий нет на указанный день
                }
                cout << endl;
            } else if (operation == "EXIT") {
                cout << "Программа завершает работу." << endl;
                return 0;
            }
        }
    }

    return 0;
}

File: zad4.py
schedule = []

while True:
    N = int(input("Введите число операций: "))

    for _ in range(N):
        operation = input("Введите операцию: ")

        if operation == "CLASS":
            day, discipline = input("Введите номер дня и название предмета: ").split()
            day = int(day)
            schedule.append((day, discipline))  # Добавляем занятие в расписание

        elif operation == "NEXT":
            last_day = 29  # Предпоследний день текущего месяца
            schedule = [(last_day, cls[1]) for cls in schedule]  # Переносим все занятия на предпоследний день
            print("Перенос всех занятий на предпоследний день текущего месяца выполнен.")

        elif operation == "VIEW":
            day = int(input("Введите номер дня: "))
            classes_on_day = [cls for cls in schedule if cls[0] == day]

            print(f"В {day} день", end=" ")
            if classes_on_day:
                print(f"1 занятие в университете: {classes_on_day[0][1]}", end="")
                for cls in classes_on_day[1:]:
                    print(f", {cls[1]}", end="")
            else:
                print("У нас свободный день!", end="")
            print()

        elif operation == "EXIT":
            print("Программа завершает работу.")
